### PR TITLE
Release Tag given better name

### PR DESCRIPTION
**Pull request by sfawcett123**
### Trello card
[CD](https://trello.com/c/UQrm3g59/516-platform-continuous-deployment)
### Context
The release name needs to tie in with the PR

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -183,29 +185,43 @@ jobs:
...
-       - name: Bump version and push tag
+       - name: Generate Tag from PR Number
          if: github.ref == 'refs/heads/master' && github.event.inputs.environment == 'Development'
          id: tag_version
-         uses: mathieudutour/github-tag-action@v5
-         with:
-           github_token: ${{ secrets.GITHUB_TOKEN }}
+         run: |
+           MATCH="Merge pull request #[0-9]+"
+           STRING=$(curl -s -X GET "https://api.github.com/repos/${{github.repository}}/commits/${{github.event.inputs.sha}}" | jq ".commit.message")
```
Comment (jebw) [inline .github/workflows/deploy.yml:208]: Doesn't this need to be `jq -r ".commit.message"` - to strip quotes out from what comes back?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -183,29 +185,43 @@ jobs:
...
+              echo "Pull request merge"
+              TAG=$( echo ${STRING} | grep -oh "[0-9]\+" | head -1 )
+              BODY=$(curl -s -X GET "https://api.github.com/repos/${REPO}/pulls/${TAG}" | jq ".body")
+           else
+              echo "Not a pull request merge"
+              TAG="0"
+              BODY="Nothing"
+           fi
+           echo ::set-output name=new_tag::"PR-${TAG}"
+           echo ::set-output name=changelog::"${BODY}"
```
Comment (jebw) [inline .github/workflows/deploy.yml:221]: Might be useful to add a step echo'ing out the outputs - could be helpful for future debugging?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -183,29 +185,43 @@ jobs:
...
-           github_token: ${{ secrets.GITHUB_TOKEN }}
+         run: |
+           MATCH="Merge pull request #[0-9]+"
+           STRING=$(curl -s -X GET "https://api.github.com/repos/${{github.repository}}/commits/${{github.event.inputs.sha}}" | jq ".commit.message")
+
+           if [[ ${STRING}  =~ ${MATCH} ]] ;
+           then
+              echo "Pull request merge"
+              TAG=$( echo ${STRING} | grep -oh "[0-9]\+" | head -1 )
+              BODY=$(curl -s -X GET "https://api.github.com/repos/${REPO}/pulls/${TAG}" | jq ".body")
```
Comment (jebw) [inline .github/workflows/deploy.yml:214]: Likewise - doesn'it it need `jq -r` to strip the quotes?

**Review by jebw — CHANGES_REQUESTED** [review]
Looks good, but from my reading it needs tweaks to the jq lines? Not actually seeing this run in order to test though?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (sfawcett123) [reply .github/workflows/deploy.yml:208]: Thanks, I will add that

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (sfawcett123) [reply .github/workflows/deploy.yml:214]: added

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (sfawcett123) [reply .github/workflows/deploy.yml:221]: will output TAG the body could be large

**Review by jebw — APPROVED** [review]
LGTM

**Pull request merged**